subsys/shell/shell.c

```diff
@@ -1145,7 +1145,7 @@ static int shell_instance_init(const struct shell *shell, const void *p_config,
 			       bool use_colors)
 {
 	assert(shell);
-	assert(shell->ctx && shell->iface && shell->name);
+	assert(shell->ctx && shell->iface && shell->prompt);
```

Comment:
what `assert` is this? I thought we used `_ASSERT()`

Reply:
I also see `_Static_assert` but we already have: `BUILD_ASSERT`

Reply:
If those are defined by the shell subsystem, could you switch to the standard Zephyr ones?

Reply:
I am using assert from here:
zephyr/lib/libc/minimal/include/assert.h
Is there any other recommended?

Reply:
I see that `assert` is the same as `__ASSERT_NO_MSG`